provider/lxd/storage.go

```diff
@@ -142,16 +166,38 @@ func (e *lxdStorageProvider) VolumeSource(cfg *storage.Config) (storage.VolumeSo
 
 // FilesystemSource is part of the Provider interface.
 func (e *lxdStorageProvider) FilesystemSource(cfg *storage.Config) (storage.FilesystemSource, error) {
-	lxdStorageConfig, err := newLXDStorageConfig(cfg.Name(), cfg.Attrs())
-	if err != nil {
-		return nil, errors.Trace(err)
+	return &lxdFilesystemSource{e.env}, nil
+}
+
+func ensureLXDStoragePool(env *environ, cfg *lxdStorageConfig) error {
+	pool, err := env.raw.StoragePool(cfg.lxdPool)
+	if err == nil {
+		// Check that the existing pool's driver
+		// and config match what we want.
+		if pool.Driver != cfg.driver {
+			return errors.Errorf(
+				`LXD storage pool %q exists, with conflicting driver %q. Specify an alternative pool name via the "lxd-pool" attribute.`,
+				pool.Name, pool.Driver,
+			)
+		}
+		for k, v := range cfg.attrs {
+			if haveV, ok := pool.Config[k]; !ok || haveV != v {
+				return errors.Errorf(
+					`LXD storage pool %q exists, with conflicting config attribute %q=%q. Specify an alternative pool name via the "lxd-pool" attribute.`,
+					pool.Name, k, haveV,
+				)
+			}
+		}
+		return nil
+	} else if !errors.IsNotFound(err) {
+		return errors.Annotatef(err, "getting storage pool %q", cfg.lxdPool)
 	}
-	return &lxdFilesystemSource{e.env, lxdStorageConfig}, nil
+	err = env.raw.CreateStoragePool(cfg.lxdPool, cfg.driver, cfg.attrs)
```

Comment:
It's not, but (a) there's no special error code, and (b) we check for existence above